src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SslStreamSecurityUpgradeProvider.cs

```diff
@@ -644,6 +618,59 @@ protected override Stream OnInitiateUpgrade(Stream stream, out SecurityMessagePr
 #endif //!FEATURE_NETNATIVE
         }
 
+        protected override async Task<Stream> OnInitiateUpgradeAsync(Stream stream, OutWrapper<SecurityMessageProperty> remoteSecurityWrapper)
+        {
```

Comment:
This class derives from StreamUpgradeInitiator which on Desktop is public abstract and has the methods sync and APN methods. While the OutWrapper mechanism is fine for internal usage, we need to find a cleaner api if we expose this. Can you open a new issue about designing a cleaner Task api for security upgrades as we probably want to expose this api at a later time.